### PR TITLE
Fixed deployment issues caused by underlying changes to foundry vm.ge…

### PR DESCRIPTION
**Pull request by mpetersoCode55**
…tCode

### CODE REVIEW

File: script/DeployAllModulesPt1.s.sol [hunk 1]
```diff
@@ -98,7 +108,7 @@ contract DeployAllModulesPt1Script is Script, DiamondScriptUtil {
             initCalldata: abi.encode(type(IDiamondInit).interfaceId)
         });
 
-        /// Deploy the diamond.
+        // /// Deploy the diamond.
```
Comment (VoR0220) [inline script/DeployAllModulesPt1.s.sol]: can we bring this to only ///?

File: script/clientScripts/DeployBase.s.sol [hunk 2]
```diff
@@ -178,9 +201,10 @@ contract DeployBase is Script, DiamondScriptUtil {
         // Loop on each facet, deploy them and create the FacetCut.
         for (uint256 facetIndex = 0; facetIndex < facets.length; facetIndex++) {
             string memory facet = facets[facetIndex];
+            string memory directory = directories[facetIndex];
 
-            // Deploy the facet.
-            bytes memory bytecode = vm.getCode(string.concat(facet, ".sol"));
+            /// Deploy the facet.
```
Comment (VoR0220) [inline script/clientScripts/DeployBase.s.sol]: ///

**Review by VoR0220 — APPROVED** [review]

File: foundry.toml [hunk 3]
```diff
@@ -2,18 +2,23 @@
...
 eth-rpc-url = 'http://anvil:8545'
+fs_permissions = [{ access = "read", path = "./out"}]
 
 [profile.deployment-check]
 eth-rpc-url = 'http://anvil-deploy-check:8545'
+fs_permissions = [{ access = "read", path = "./out"}]
 
 [profile.polygon]
 eth-rpc-url = 'http://172.20.0.2:10002'
+fs_permissions = [{ access = "read", path = "./out"}]
```
Comment (VoR0220) [inline foundry.toml:21]: Can we get an explanation on how this works or a link to it? I imagine this is the breaking change right here.

Comment (mpetersoCode55) [reply foundry.toml:21]: So essentially vs.getCode no longer pulls from filesystem based solidity files but instead pulls from the compiled json output in the out directory.
The foundry vm calls do not have read access to the out directory by default however, so we had to add this line to our foundry.toml on top of refactoring how we grab the files with vm.getCode

File: script/clientScripts/DeployBase.s.sol (hunk 2 above)
Comment (mpetersoCode55) [reply script/clientScripts/DeployBase.s.sol]: Good catch!

File: script/DeployAllModulesPt1.s.sol (hunk 1 above)
Comment (mpetersoCode55) [reply script/DeployAllModulesPt1.s.sol]: Good catch!

**Review by excalq — CHANGES_REQUESTED** [review]
Posting details of why in a minute...

File: foundry.toml (hunk 3 above)
Comment (VoR0220) [reply foundry.toml:21]: Very useful knowledge. Hope they updated their docs to accompany this.

File: install-deploy-check.sh [hunk 4]
```diff
@@ -8,11 +8,8 @@ python3 -m venv .venv
 source .venv/bin/activate
 python3 -m pip install -r requirements.txt
 
-if [ $SCRIPT_MODE = "--with-build" ]; then
-  forge build   
-fi
-
 if [ $SCRIPT_MODE = "--with-deploy-check" ]; then
+  forge build
```
Comment (excalq) [inline install-deploy-check.sh]: [Dockerfile.tron](https://github.com/thrackle-io/tron/blob/main/Dockerfile.tron#L13-L15) needs to run build exclusively during the image build (the `RUN` step), and not on container startup (the `CMD` step). This causes every startup of the container to re-compile.

**Review by excalq — CHANGES_REQUESTED** [review]

File: install-deploy-check.sh (hunk 4 above)
Comment (mpetersoCode55) [reply install-deploy-check.sh]: That makes sense, I'll set this back. I assume the build version is always run before any form of deployment? Deploying no longer works without an explicit build first

File: install-forge.sh [hunk 5]
```diff
@@ -9,11 +9,8 @@ python3 -m venv .venv
 source .venv/bin/activate
 python3 -m pip install -r requirements.txt
 
-if [ $SCRIPT_MODE = "--with-build" ]; then
-	forge build
-fi
-
 if [ $SCRIPT_MODE = "--with-deploy" ]; then
```
Comment (excalq) [inline install-forge.sh]: Here as well

File: foundry.toml (hunk 3 above)
Comment (mpetersoCode55) [reply foundry.toml:21]: They did not, all of their examples in their docs for vs.getCode no longer work

File: install-deploy-check.sh (hunk 4 above)
Comment (excalq) [reply install-deploy-check.sh]: Yes, the Dockerfile always will run `RUN ./install-forge.sh --with-build` when the image is created. Deployment or any restart of the workload in K8s (or in Docker-Compose) will run `./install-forge.sh --with-deploy` at startup. (Maybe that could be better named?)

File: foundry.toml (hunk 3 above)
Comment (excalq) [reply foundry.toml:21]: Might be worth keeping a comment above the first use of it, with that info.

**Review by excalq — APPROVED** [review]
Thanks for the deployment fix

**Pull request merged**